contracts/ColonyTask.sol

```diff
@@ -402,10 +517,12 @@ contract ColonyTask is ColonyStorage, DSMath {
 
   // Get the function signature and task id from the transaction bytes data
   // Note: Relies on the encoded function's first parameter to be the uint256 taskId
-  function deconstructCall(bytes _data) internal pure returns (bytes4 sig, uint256 taskId) {
+  function deconstructCall(bytes _data) internal pure returns (bytes4 sig, uint256 taskId, uint8 roleId, address userAddress) {
     assembly {
       sig := mload(add(_data, 0x20))
-      taskId := mload(add(_data, add(0x20, 4))) // same as calldataload(72)
+      taskId := mload(add(_data, 0x24)) // same as calldataload(72)
+      roleId := mload(add(_data, 0x44))
+      userAddress := mload(add(_data, 0x64))
```

Comment:
`roleId` and  `userAddress ` won't work for anything but the `setTaskRoleUser` function. This feels very wrong..